sklearn/preprocessing/_polynomial.py

```diff
@@ -958,14 +990,23 @@ def transform(self, X):
         """
         check_is_fitted(self)
 
-        X = self._validate_data(X, reset=False, accept_sparse=False, ensure_2d=True)
+        X = self._validate_data(
+            X,
+            reset=False,
+            accept_sparse=False,
+            ensure_2d=True,
+            force_all_finite=False,
+        )
 
         n_samples, n_features = X.shape
         n_splines = self.bsplines_[0].c.shape[1]
         degree = self.degree
 
+        # get indicator for nan values
+        nan_indicator = _get_mask(X, np.nan)
```

Comment:
Yes because we only allocate a temporary array of shape `(n_samples,)` at a time instead an `n_features` times bigger array of shape `(n_samples, n_features)` ahead of time.